scrapy/linkextractors/htmlparser.py

```diff
@@ -39,7 +40,7 @@ def _extract_links(self, response_text, response_url, response_encoding):
         ret = []
         base_url = urljoin(response_url, self.base_url) if self.base_url else response_url
         for link in links:
-            if isinstance(link.url, unicode):
+            if isinstance(link.url, six.text_type):
```

Comment:
The main reason for making URLs native strings is that this is how Python standard libraries work. In 2.x url-related libraries want urls as bytes, in Python 3.x url-related libraries want them as unicode. Also, it is cumbersome to work with byte URLs in Python 3 (no string formatting, etc.), and it is cumbersome and backwards incompatible to switch to unicode URLs in Python 2.x.